### ISSUE TITLE
make errors on Ubuntu 18.04.3 

### ISSUE THREAD

**Issue by tehNYAZ** [jichu4n/jfbview#30]
Aside from g++, I've also tried:
  * compiling with the latest version of gcc and clang.
  * using `CXXFLAGS="Wno-format-security"`.
```bash
git-cloned$ sudo apt-get install \
>   libssl-dev \
>   libopenjp2-7-dev \
>   libjbig2dec-dev \
>   libharfbuzz-dev \
>   libncurses5-dev \
>   libimlib2-dev \
>   libmupdf-dev
Reading package lists... Done
Building dependency tree
Reading state information... Done
Note, selecting 'libjbig2dec0-dev' instead of 'libjbig2dec-dev'
libharfbuzz-dev is already the newest version (1.7.2-1ubuntu1).
libjbig2dec0-dev is already the newest version (0.13-6).
libimlib2-dev is already the newest version (1.4.10-1).
libncurses5-dev is already the newest version (6.1-1ubuntu1.18.04).
libssl-dev is already the newest version (1.1.1-1ubuntu2.1~18.04.4).
libopenjp2-7-dev is already the newest version (2.3.0-2build0.18.04.1).
libmupdf-dev is already the newest version (1.14.0-0build3+ubuntu18.04).
0 upgraded, 0 newly installed, 0 to remove and 5 not upgraded.
git-cloned$ git clone https://github.com/jichu4n/jfbview.git
Cloning into 'jfbview'...
remote: Enumerating objects: 1334, done.
remote: Total 1334 (delta 0), reused 0 (delta 0), pack-reused 1334
Receiving objects: 100% (1334/1334), 309.16 KiB | 3.06 MiB/s, done.
Resolving deltas: 100% (943/943), done.
git-cloned$ cd jfbview
jfbview$ make
g++ -Wall -O2 -std=c++1y -o mupdf_version mupdf_version.cpp  -lmupdf

=============================
Detected configuration:

MUPDF_VERSION = 10016
OPENJP2 = openjp2
=============================

make[1]: Entering directory '/home/tehnyaz/git-cloned/jfbview'
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o command.o command.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o document.o document.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o framebuffer.o framebuffer.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o image_document.o image_document.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o multithreading.o multithreading.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o outline_view.o outline_view.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o pdf_document.o pdf_document.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o pixel_buffer.o pixel_buffer.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o search_view.o search_view.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o string_utils.o string_utils.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o ui_view.o ui_view.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o viewer.o viewer.cpp
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016   -c -o main.o main.cpp
main.cpp: In function ‘int main(int, char**)’:
main.cpp:716:47: warning: format not a string literal and no format arguments [-Wformat-security]
     fprintf(stderr, FRAMEBUFFER_ERROR_HELP_STR);
                                               ^
g++ -Wall -O2 -std=c++1y -DMUPDF_VERSION=10016 -o jfbview command.o document.o framebuffer.o image_document.o multithreading.o outline_view.o pdf_document.o pixel_buffer.o search_view.o string_utils.o ui_view.o viewer.o main.o  -lpthread -lform -lncurses -lmupdf -lfreetype -lharfbuzz -lz -ljbig2dec -ljpeg -lopenjp2 -lmupdf-third -lssl -lcrypto -lImlib2
pdf_document.o: In function `PDFDocument::Open(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)':
pdf_document.cpp:(.text+0x13da): undefined reference to `fz_do_catch'
pdf_document.cpp:(.text+0x141d): undefined reference to `fz_do_try'
collect2: error: ld returned 1 exit status
Makefile:101: recipe for target 'jfbview' failed
make[1]: *** [jfbview] Error 1
make[1]: Leaving directory '/home/tehnyaz/git-cloned/jfbview'
Makefile:125: recipe for target 'all' failed
make: *** [all] Error 2
```
edit: formatting.

**Comment by jichu4n** [comment]
Thanks for the feedback!

I haven't been able to repro this error. At first glance, my suspicion is that we may have a problem with detecting the MuPDF version. At the beginning of the output, we see that the version of MuPDF installed is 1.14:

> libmupdf-dev is already the newest version (1.14.0-0build3+ubuntu18.04).

But later on in the build process:

>Detected configuration:
>
>MUPDF_VERSION = 10016

...which corresponds to MuPDF 1.16. This is quite strange. Some ideas:

- Is is possible that you have MuPDF 1.16 installed somewhere?
   * Could you try running the following to check which MuPDF installation is being used? This should normally return `/usr/include/mupdf/fitz.h`.
     ` g++ -H -std=c++1y mupdf_version.cpp 2> >(grep fitz\\.h)`
    * If so, could you check if the file is somehow provided by a different package? e.g. does `dpkg -S /usr/include/mupdf/fitz.h` return `libmupdf-dev`?
- Is it possible that the `mupdf_version` binary is from a different build? e.g. try `make clean && make`?

**Comment by tehNYAZ** [comment]
Hey thanks so much for the troubleshooting tips you gave.

```g++ -H -std=c++1y mupdf_version.cpp 2> >(grep fitz\\.h)```
Returned:
```/usr/include/mupdf/fitz.h```

```dpkg -S /usr/include/mupdf/fitz.h```
did return:
```dpkg -S /usr/include/mupdf/fitz.h```

Based on what you said, I figured out that I had mupdf 1.16 compiled from source at some point.
```dpkg -L mupdf``` and ```apt show mupdf``` couldn't find the package.
However, I was able to find directory of the git repo I cloned to compile v. 1.16 and was able to use ```make -n install``` to cleanup after ```make```  since there was no ```uninstall``` target in its ```Makefile```.
I did end up removing and reinstalling ```libmupdf-dev``` because fitz got removed at this point (might have been my fault).

I should really use [checkinstall](http://manpages.ubuntu.com/manpages/trusty/man8/checkinstall.8.html) with the ```-D``` option next time instead of ```make``` for compiling from source.

Anyhow, thanks for your help.
